src/Orleans/LogConsistency/ILogConsistencyProtocolServices.cs

```diff
@@ -0,0 +1,145 @@
+﻿
+using System;
+using System.Net;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Orleans.Providers;
+using Orleans.Runtime;
+using Orleans.MultiCluster;
+using Orleans.GrainDirectory;
+
+namespace Orleans.LogConsistency
+{
+    /// <summary>
+    /// Functionality for use by log view adaptors that use custom consistency or replication protocols.
+    /// Abstracts communication between replicas of the log-consistent grain in different clusters.
+    /// </summary>
+    public interface ILogConsistencyProtocolServices
+    {
+        /// <summary>
+        /// Send a message to a remote cluster.
+        /// </summary>
+        /// <param name="payload">the message</param>
+        /// <param name="clusterId">the destination cluster id</param>
+        /// <returns></returns>
+        Task<ILogConsistencyProtocolMessage> SendMessage(ILogConsistencyProtocolMessage payload, string clusterId);
+
+
+        /// <summary>
+        /// The untyped reference for this grain.
+        /// </summary>
+        GrainReference GrainReference { get;  }
+
+        
+        /// <summary>
+        /// The multicluster registration strategy for this grain.
+        /// </summary>
+        IMultiClusterRegistrationStrategy RegistrationStrategy { get; }
+
+
+        /// <summary>
+        /// Whether this cluster is running in a multi-cluster network.
+        /// </summary>
+        /// <returns></returns>
+        bool MultiClusterEnabled { get; }
+
+        // If there is no multi-cluster network,
+        // we use a default multi-cluster configuration containing just one cluster (this one), named "I"
+        // this is more convenient and consistent than returning null or invalid,
+        // since it means the log-consistency providers do the right thing (run like a single-cluster configuration)
+
+        /// <summary>
+        /// The id of this cluster. Returns "I" if no multi-cluster network is present.
+        /// </summary>
+        /// <returns></returns>
+        string MyClusterId { get; }
+
```

Comment:
Looks like comments got lost here because of file renaming.
@jdom
>Do we really need both ClusterId and DeploymentId as different values? Maybe we should consolidate? DeploymentId is really just a logical name for the cluster, it's just that it typically maps to the cloud service's deployment id when deploying to it, but it's an awful name when used in a different environment, as it doesn't map to its logical intended purpose

@sebastianburckhardt 
>I think DeploymentId has an Azure-specific meaning. So, I don't think they are the same. ClusterId is a logical name for the cluster (e.g., "uswest", "europe", "front1", "front2"). DeploymentId is something like "a74ff75b9bd6424ebbff3bde9643074b" and changes everytime I redeploy the service.

@dom
>That's exactly why I say it's an awful name, as it implies the wrong meaning. DeploymentId is the cluster name. It's just that when you deploy to azure cloud services (not even other flavors of azure), it picks its DeploymentId as the cluster name by default (and only if unset). Nevertheless when deploying in a different way (on premises, other clouds, other azure services, yams, etc) you must set the DeploymentId to a logical name explicitly, because it is in fact the cluster id.

Yes, ClusterId and DeploymentId are really synonyms. Although in the hindsight, ClusterId is probably a better name. But it's too late to change it in our existing APIs. However, if here MyClusterId is contained within the log consistency provider scope, and within the geo-distribution context ClusterId makes more sense in conjunction with the notion of multi-cluster, I don't see a problem using `MyClusterId` within this context. Unless I missed some other implications.